.github/workflows/create-release.yml

```diff
@@ -45,23 +57,52 @@ jobs:
         echo "# Platform-specific torch requirements (See SCT Issue #2745)" >> requirements-freeze.txt
         grep "torch" requirements.txt >> requirements-freeze.txt
 
-    - name: Update version.txt
+    - name: Update version.txt (using test value for CI)
+      run: |
+        echo "0.0" > spinalcordtoolbox/version.txt
+      # Only attach to release if workflow is run manually. (This allows the workflow to double as a PR test.)
+      if: github.event_name != 'workflow_dispatch'
+
+    - name: Update version.txt (using milestone title)
       run: |
         echo "${{ github.event.inputs.milestone_title }}" > spinalcordtoolbox/version.txt
+      # Only attach to release if workflow is run manually. (This allows the workflow to double as a PR test.)
+      if: github.event_name == 'workflow_dispatch'
+
+    - name: Reinstall SCT from release branch
+      run: |
+        ./install_sct -yc
+        cat ~/.bashrc | grep "export SCT_DIR" | cut -d " " -f 2 >> $GITHUB_ENV
+        cat ~/.bashrc | grep "export PATH" | grep -o "/.*" | cut -d ':' -f 1 >> $GITHUB_PATH
+      # NB: install_sct edits ~/.bashrc, but those environment changes don't get passed to subsequent steps in GH Actions.
+      # So, we filter through the .bashrc and pass the values to $GITHUB_ENV and $GITHUB_PATH.
+      # Relevant documentation: https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#environment-files
+      # This workaround should be replaced by https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3198#discussion_r568225392
+
+    - name: Run sct_check_dependencies
+      run: |
+        sct_check_dependencies
+
+    - name: Run sct_testing
+      run: |
+        sct_testing
```

Comment:
Hmm. I'm surprised that https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3672 isn't triggered by this CI run. I assumed that if `sct_testing` was called post-install from `release`, then this issue would crop up. :thinking: 

I'll look into this and see what I can find.

Reply:
Update: While I'm not sure if this is the cause of the issue, I did find one bug in `install_sct`. Currently, to determine whether this is a `release` installation, we use the following logic: 

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/f068452c43e19669faca45c222f3d6fca60ca743/install_sct#L412-L420

Checking for `.git` works if you're downloading the source code from the Releases page, but _doesn't_ work if you've checked out the `release` branch (which we do in this GitHub Workflow). So, I think it might be more reliable to check for the presence of `requirements-freeze.txt`, which is the only release-specific file.

This could have some effect on whether the tests are truly being run from the "release" branch as I hoped.

Reply:
Ah, and fixing the problem in this branch does nothing, because `create-release.yml` checks out `master` first. :laugh:

Reply:
Excellent! I switched the main branch to this PR, and now the test fails as expected.

This gives us some assurance that #3672 would have been caught by this worklflow, and that we can catch any other release-related issues ahead of time.

Reply:
I think I'm going to test a fix for #3672 in this branch for convenience sake, since we have the CI ready for debugging purposes, and merging this PR as-is will just cause a failure on `master`.

Reply:
Ok, I've got a fix up in [93cc464](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3678/commits/93cc4640faf39de501baf933f292ef0b69667a17). :)